Sources/AsyncHTTPClient/ConnectionPool/HTTP2/HTTP2ClientRequestHandler.swift

```diff
@@ -295,6 +322,36 @@ final class HTTP2ClientRequestHandler: ChannelDuplexHandler {
         }
     }
 
+    private func runTimeoutAction(_ action: IdleWriteStateMachine.Action, context: ChannelHandlerContext) {
+        switch action {
+        case .startIdleWriteTimeoutTimer(let timeAmount):
+            assert(self.idleWriteTimeoutTimer == nil, "Expected there is no timeout timer so far.")
+
+            self.idleWriteTimeoutTimer = self.eventLoop.scheduleTask(in: timeAmount) {
+                guard self.idleWriteTimeoutTimer != nil else { return }
```

Comment:
In HTTP1 we use a timer "ID" but here we are just setting the timer to nil. Is there a reason for these two different approaches?

Reply:
Yes, this is an inconsistency and should be changed because race conditions can happen with this current approach. It was actually fixed for HTTP1 in https://github.com/swift-server/async-http-client/pull/455, but not in HTTP2. I just decided to keep things as they currently are in this PR (without using the timer ID) to avoid mixing multiple changes in a single PR, but I'll send a follow-up that adds a timer and changes these checks.